assets/scripts/info_bubble/RemoveButton.jsx

```diff
@@ -27,30 +20,36 @@ class RemoveButton extends React.PureComponent {
       trackEvent('INTERACTION', 'REMOVE_ALL_SEGMENTS', 'BUTTON', null, true)
     } else {
       // Otherwise, remove one segment
-      removeSegment(this.props.segment)
+      removeSegment(props.segment)
       trackEvent('INTERACTION', 'REMOVE_SEGMENT', 'BUTTON', null, true)
     }
   }
 
-  render () {
-    // Bail if segment is not provided; do not check for falsy. 0 is valid value for segment
-    if (typeof this.props.segment === 'undefined' || this.props.segment === null) return null
-
-    return (
-      <button
-        className="info-bubble-remove"
-        tabIndex={-1}
-        title={this.props.intl.formatMessage({ id: 'tooltip.remove-segment', defaultMessage: 'Remove segment' })}
-        onClick={this.onClick}
-      >
-        <FontAwesomeIcon
-          icon={ICON_TRASH}
-          className="remove-icon"
-        />
-        <FormattedMessage id="btn.remove" defaultMessage="Remove" />
-      </button>
-    )
-  }
+  // Bail if segment is not provided; do not check for falsy. 0 is valid value for segment
+  if (typeof props.segment === 'undefined' || props.segment === null) return null
+
+  return (
+    <button
+      className="info-bubble-remove"
+      tabIndex={-1}
+      title={intl.formatMessage({ id: 'tooltip.remove-segment', defaultMessage: 'Remove segment' })}
+      onClick={onClick}
+    >
+      <FontAwesomeIcon
+        icon={ICON_TRASH}
+        className="remove-icon"
+      />
+      <FormattedMessage id="btn.remove" defaultMessage="Remove" />
+    </button>
+  )
+}
+
+RemoveButton.propTypes = {
+  segment: PropTypes.number.isRequired
+}
+
+RemoveButton.defaultProps = {
```

Comment:
Move `propTypes` above component declaration and destructure `defaultProps` within the component

Reply:
Corrected.
Please check if I understood correctly regarding destructure defaultProps :)